src/index.ts

```diff
@@ -38,20 +39,23 @@ export const getPaginationState = ({
     };
 };
 
-type CurrentPageReducerActions = {| type: "SET", page: number |} | {| type: "NEXT" | "PREV" |};
+type CurrentPageReducerActions = React.Reducer<
+    number,
+    Partial<{ type: "SET"; page: number } | { type: "NEXT" | "PREV" }>
```

Comment:
I'm worried the `Partial` here will cause all fields to be optional, when really we want to preserve this as a discriminated union type

Reply:
Good point! The current union works well, but how should `action = {}` in the reducer be initialized? With the union, `type` must be: `"SET" | "NEXT" | "PREV"`